universite-du-quebec-a-montreal.csl

```diff
@@ -564,51 +756,127 @@
             <text macro="translation"/>
           </group>
         </group>
+        <choose>
+          <if type="chapter" match="any">
+            <text variable="number" prefix=", "/>
+          </if>
+        </choose>
+        <choose>
+          <if type="article book dataset" match="any">
+            <choose>
+              <if match="any" variable="author editor">
+                <text variable="genre" prefix=" [" suffix="]"/>
+              </if>
+              <else>
+                <text variable="genre" prefix=". [" suffix="]"/>
+              </else>
+            </choose>
+            <text variable="collection-title"/>
+          </if>
+        </choose>
       </else-if>
-      <else-if type="graphic motion_picture report song paper-conference" match="any">
+      <else-if type="graphic motion_picture song paper-conference musical_score" match="any">
         <group delimiter=", " prefix=" (" suffix=")">
           <text macro="edition"/>
           <group>
-            <text term="volume" form="short" plural="true" text-case="capitalize-first" suffix=" "/>
-            <number variable="number-of-volumes" form="numeric" prefix="1-"/>
-          </group>
-          <group>
-            <text term="volume" form="short" text-case="capitalize-first" suffix=" "/>
+            <text term="volume" form="short" suffix=" "/>
             <number variable="volume" form="numeric"/>
           </group>
           <group>
             <label suffix=" " variable="page" form="short"/>
             <text variable="page"/>
           </group>
         </group>
+        <choose>
+          <if type="song musical_score" match="any">
+            <choose>
+              <if match="any" variable="container-title">
+                <text variable="medium" prefix=" [" suffix="]"/>
+                <text variable="genre" prefix=" [" suffix="]"/>
+                <text variable="number" prefix=" (épisode " suffix=")"/>
```

Comment:
ditto on the concern with episode

Reply:
As said above, this is based on APA style blog. I would like to keep it if you agree.